cmd/snap-confine/mount-support.c

```diff
@@ -333,11 +334,34 @@ static void sc_bootstrap_mount_namespace(const struct sc_mount_config *config)
 						 config->rootfs_dir, dir);
 				sc_must_snprintf(dst, sizeof dst, "%s%s",
 						 scratch_dir, dir);
-				sc_do_mount(src, dst, NULL, MS_BIND, NULL);
-				sc_do_mount("none", dst, NULL, MS_SLAVE, NULL);
+				if (access(src, F_OK) == 0
```

Comment:
We should be extra careful with symlinks here (the base snap may contain symlinks that will "pass" the F_OK test. I'd suggest using `lstat` instead, which should not be much longer but will give us the predictability.

Reply:
I'm find with lstat(), I kept it in sync with the lines before that use access() as well.

Reply:
Changed now